From ad9fb985df470bed5f77da4701f2aebe45af5ff3 Mon Sep 17 00:00:00 2001
From: Mill Chen <millchen@google.com>
Date: Mon, 23 Dec 2024 09:54:36 -0800
Subject: [PATCH] Add default Sound Picker package name

To avoid implicit intents, make intents launch explicitly.

DISABLE_TOPIC_PROTECTOR

Bug: 333344157
Test: Build
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:255615873996b5282d3d3709904eb931737ab358)
Merged-In: I905c0f3ddaf0f488ff2fc3bf1747b6df0d802504
Change-Id: I905c0f3ddaf0f488ff2fc3bf1747b6df0d802504
---
 res/values/config.xml                            | 5 +++++
 src/com/android/settings/RingtonePreference.java | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/res/values/config.xml b/res/values/config.xml
index 334d4e574c..6c2f4e3bd6 100755
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -272,6 +272,11 @@
     <!-- Whether Settings intelligence supports presenting Slices. -->
     <bool name="config_settingsintelligence_slice_supported">false</bool>
 
+    <!-- Sound picker package name -->
+    <string name="config_sound_picker_package_name" translatable="false">
+        com.android.soundpicker
+    </string>
+
     <!-- AOSP Emergency app package name -->
     <string name="config_aosp_emergency_package_name" translatable="false">
         com.android.emergency
diff --git a/src/com/android/settings/RingtonePreference.java b/src/com/android/settings/RingtonePreference.java
index de5b7c39f7..6b9dad7ffb 100644
--- a/src/com/android/settings/RingtonePreference.java
+++ b/src/com/android/settings/RingtonePreference.java
@@ -74,7 +74,8 @@ public class RingtonePreference extends Preference {
                 true);
         mShowSilent = a.getBoolean(com.android.internal.R.styleable.RingtonePreference_showSilent,
                 true);
-        setIntent(new Intent(RingtoneManager.ACTION_RINGTONE_PICKER));
+        setIntent(new Intent(RingtoneManager.ACTION_RINGTONE_PICKER)
+                .setPackage(context.getString(R.string.config_sound_picker_package_name)));
         setUserId(UserHandle.myUserId());
         a.recycle();
     }
-- 
2.48.1.262.g85cc9f2d1e-goog

